File: back/app/schemas/conversation.py
"""DTOs da tela de monitoramento do agente.

O painel só lê estas tabelas (quem escreve nelas é o Agente 2); a única escrita
exposta é o toggle de handoff, que pausa o bot para um humano assumir.
"""

from __future__ import annotations

from datetime import datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, Field

from app.domain.enums import MessageDirection
from app.schemas.common import ORMModel


class ConversationRead(ORMModel):
    id: UUID
    phone: str
    channel: str
    state: str
    handoff: bool
    fail_count: int
    customer_name: str | None = None
    last_message_preview: str | None = None
    active_order_id: UUID | None = None
    last_message_at: datetime | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None


class ConversationMessageRead(ORMModel):
    id: UUID
    direction: MessageDirection
    content: str
    state_before: str | None = None
    state_after: str | None = None
    detected_intent: str | None = None
    confidence: Decimal | None = None
    llm_model: str | None = None
    created_at: datetime


class HandoffUpdate(BaseModel):
    """Corpo de `POST /api/conversations/{id}/handoff`."""

    handoff: bool


class HandoffResult(BaseModel):
    id: UUID
    handoff: bool
    state: str


class ConversationList(BaseModel):
    conversations: list[ConversationRead] = Field(default_factory=list)


__all__ = [
    "ConversationList",
    "ConversationMessageRead",
    "ConversationRead",
    "HandoffResult",
    "HandoffUpdate",
]

File: front/src/types/conversation.ts
/**
 * Tipos da tela de monitoramento do agente de WhatsApp.
 * Os estados espelham `back/app/domain/enums.py::ConversationState`.
 */

export const CONVERSATION_STATES = [
  'conversando',
  'confirmando_pedido',
  'aguardando_pagamento',
  'concluido',
  'atendimento_humano',
  'cancelado',
] as const;

export type ConversationState = (typeof CONVERSATION_STATES)[number];

export type MessageDirection = 'entrada' | 'saida';

export interface Conversation {
  id: string;
  phone: string;
  /** String livre no banco; tratamos valores desconhecidos sem quebrar a tela. */
  state: ConversationState | string;
  handoff: boolean;
  channel?: string;
  customer_name?: string | null;
  last_message_at?: string | null;
  last_message_preview?: string | null;
  active_order_id?: string | null;
  fail_count?: number;
}

export interface ConversationMessage {
  id?: string;
  direction: MessageDirection;
  content: string;
  detected_intent?: string | null;
  state_before?: string | null;
  state_after?: string | null;
  created_at: string;
}
